packages/frontend/src/components/SimpleTable/DashboardCellContextMenu.tsx

```diff
@@ -59,34 +58,36 @@ const DashboardCellContextMenu: FC<
     const filterField =
         isDimension(item) && !item.hidden
             ? [
-                  {
-                      id: uuid4(),
-                      target: {
-                          fieldId: fieldId(item),
-                          tableName: item.table,
-                      },
-                      operator: FilterOperator.EQUALS,
-                      values: [value.raw],
-                      label: undefined,
-                  },
+                  createDashboardFilterRuleFromField({
+                      field: item,
+                      availableTileFilters: {},
+                      isTemporary: true,
+                      value: value.raw,
+                  }),
               ]
             : [];
 
     const possiblePivotFilters = (
         meta?.pivotReference?.pivotValues || []
-    ).map<DashboardFilterRule>((pivot) => {
+    ).reduce<DashboardFilterRule[]>((acc, pivot) => {
         const pivotField = itemsMap?.[pivot?.field];
-        return {
-            id: uuid4(),
-            target: {
-                fieldId: pivot.field,
-                tableName: isField(pivotField) ? pivotField?.table : '',
-            },
-            operator: FilterOperator.EQUALS,
-            values: [pivot.value],
-            label: undefined,
-        };
-    });
+        if (
+            !pivotField ||
+            !isField(pivotField) ||
+            !isFilterableField(pivotField)
+        )
+            return acc;
+
+        acc.push(
+            createDashboardFilterRuleFromField({
+                field: pivotField,
+                availableTileFilters: {},
+                isTemporary: true,
+                value: pivot.value,
+            }),
+        );
+        return acc;
```

Comment:
nit: this can be `return [...acc, createDashboarblabla]` to avoid using push

Reply:
sure!